framework/infrastructure/gcp/api.py

```diff
@@ -520,19 +530,25 @@ def parent(self, location: Optional[str] = GLOBAL_LOCATION):
             location = self.GLOBAL_LOCATION
         return f"projects/{self.project}/locations/{location}"
 
-    def resource_full_name(self, name, collection_name):
-        return f"{self.parent()}/{collection_name}/{name}"
+    def resource_full_name(
+        self, name, collection_name, location: Optional[str] = None
```

Comment:
Instead of `Optional[str] = None`, assign the default value to `GLOBAL_LOCATION`:
```suggestion
        self, name, collection_name, location: str = GLOBAL_LOCATION
```